urle381b6cba7_IkarosKappler_ikrs_http_tgz-pJ8-ikrs_json_rpc_JSONRPCHandlerJ8/src/ikrs/httpd/resource/AbstractDirectoryResource.java

```diff
@@ -0,0 +1,330 @@
+import java.io.InputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.URI;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.UUID;
+import java.util.logging.Level;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock.ReadLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock.WriteLock; 
+
+
+import ikrs.httpd.HTTPFileFilter;
+import ikrs.httpd.HTTPHandler;
+import ikrs.httpd.ReadOnlyException;
+import ikrs.httpd.Resource;
+import ikrs.util.CaseInsensitiveComparator;
+import ikrs.util.CustomLogger;
+import ikrs.util.MIMEType;
+
+/**
+ * @author   Ikaros Kappler
+ * @date     2012-07-20
+ * @modified 2012-10-11
+ * @version  1.0.0
+ **/
+
+public abstract class AbstractDirectoryResource
+    extends AbstractResource 
+    implements DirectoryResource {
+
+    /**
+     * The actual directory (SHOULD be a directory but that's not guaranteed).
+     **/
+    private File dir;
+
+    /**
+     * The URI the directory file was requested through.
+     **/
+    private URI requestURI;
+
+
+    /**
+     * The current session's ID.
+     **/
+    private UUID sessionID;
+
+    /**
+     * The input stream that will be used as a buffer for the generated contents.
+     **/
+    private ByteArrayInputStream inputStream;
+
+    /**
+     * The length of the generated contents (will be set after content generation).
+     **/
+    private long length;
+
+    /**
+     * The date format to use.
+     **/
+    private DateFormat dateFormat;
+
+    /**
+     * The file filter to use.
+     **/
+    private HTTPFileFilter fileFilter;
+    
+    /**
+     * Create a new AbstractDirectoryResource.
+     *
+     * @param logger       A custom logger to write log messages to.
+     * @param fileFilter   The file filter to use.
+     * @param dir          The directory.
+     * @param requestURI   The uri from the request (will be used to avoid printing the absolute file path).
+     * @param useFairLocks If set to true the created resource will use fair locks.
+     **/
+    public AbstractDirectoryResource( HTTPHandler handler,
+				      CustomLogger logger,
+				      HTTPFileFilter fileFilter,
+				      File dir,
+				      URI requestURI,
+				      UUID sid,
+				      boolean useFairLocks ) 
+	throws NullPointerException {
+
+	super( handler, logger, useFairLocks );
+
+	if( dir == null )
+	    throw new NullPointerException( "Cannot create AbstractDirectoryResource with null-directory." );
+       
+	if( requestURI == null )
+	    throw new NullPointerException( "Cannot create AbstractDirectoryResource with null-URI." );
+
+	this.fileFilter   = fileFilter;
+	this.dir          = dir;
+	this.requestURI   = requestURI;
+	this.sessionID    = sid;
+
+	this.dateFormat   = new SimpleDateFormat( "yyyy-MM-dd HH:mm:ss" );
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The SimpleDateFormat object was created without setting the time zone. Make sure that you want to use the default time zone or use letter z, Z or X in the pattern. Otherwise, if you do not call setTimeZone() on the created SimpleDateFormat object, you might get an unexpected date and time when using the object.